analyzers/src/SonarAnalyzer.CSharp/Extensions/ISymbolExtensions.cs

```diff
@@ -89,5 +91,33 @@ private static IEnumerable<SyntaxNode> GetDescendantNodes(Location location, Syn
 
             return root.DescendantNodes();
         }
+
+        public static SyntaxToken? FirstDeclaringReferenceIdentifier(this ISymbol symbol) =>
+            symbol.DeclaringReferenceIdentifiers().FirstOrDefault();
+
+        public static ImmutableArray<SyntaxToken> DeclaringReferenceIdentifiers(this ISymbol symbol) =>
+            symbol.DeclaringSyntaxReferences
+               .Select(x => x.GetSyntax().NodeIdentifier())
+               .WhereNotNull()
+               .ToImmutableArray();
+
+        public static SyntaxToken? NodeIdentifier(this SyntaxNode node)
+            => node.RemoveParentheses() switch
```

Comment:
Formatting nitpick `=>`
```suggestion
        public static SyntaxToken? NodeIdentifier(this SyntaxNode node) => 
            node.RemoveParentheses() switch
```